src/jsx.ts

```diff
@@ -27,17 +27,22 @@ function flattenAndFilter (children: JsxVNodeChildren[], flattened: VNode[]): VN
  * jsx/tsx compatible factory function
  * see: https://www.typescriptlang.org/docs/handbook/jsx.html#factory-functions
  */
-export function jsx (tag: string | FunctionComponent, data: VNodeData | null, ...children: JsxVNodeChildren[]): VNode {
+export function jsx (tag: string | FunctionComponent, props: JsxVNodeProps | null, ...children: JsxVNodeChildren[]): VNode {
   const flatChildren = flattenAndFilter(children, []);
   if (typeof tag === 'function') {
     // tag is a function component
-    return tag(data, flatChildren);
+    return tag(props, flatChildren);
   } else {
+    // append sel css selector to tag to support equivalent of h('span.foo.bar')
+    if (props && props.sel) {
+      tag += props.sel;
+      props.sel = undefined;
```

Comment:
While this does not delete the property, it still mutates the provided object. Is there a problem with leaving it untouched?

Reply:
we could, but the test diff looks off because the vnode has {sel, props: {sel...}, children: ..}

I could make another object but that seemed like unnecessary perf hit.

usually for jsx, the props are generated by the transpiler which would end up as vnode objects anyway so it seemed okay modifying them before passing to underlying h which would generate the vnode.

Happy to change it if you feel really strongly, but I didn't consider mutating sel to be a no-no since it is a jsx specific key only.

Reply:
I suppose leaving it as it is would be the least performance hit. And how the tests look doesn't seem to be a reason to change implementation, to me. I don't know if this is a hot path or not. I think it might be. So maybe we should see that as a reason to not mutate.